### PR TITLE
Hai Reveal conversations: bug fix to log entries, remove looping dialog and action choices from part 6

### PR DESCRIPTION
**Pull request by UnorderedSigh**
**Feature:** This PR uses `to display` to get rid of looping dialog, removes action choices, and fixes some minor conversation bugs, in part 6.

This PR is not as long as it looks. The line count comes from technical stuff (to display, set, has, etc.) or splitting up paragraphs.

Note: My dialog cleanup work has been split into smaller PRs by @MasterOfGrey's request, to make them easier to review. This PR deals with part 6.

## Feature Details
This cleans up dialog in Hai Reveal part 6 by:

1. Replacing action choices with dialog choices, which makes the story more engaging.
2. Adding a second dialog option, so the dialog choice doesn't seem like a "(Continue.)" option.
3. Removing looping dialog.

### Bug Fixes to Terry Log Entries

@bene-dictator discovered that the Terry killed/spared log entries didn't show up. 

They didn't show up because they relied on an ": on offer" variable being set by another mission whose "on offer" has a conversation. Those variables aren't set until the player lands again. The invisible mission that set that log only triggers on Aspiration, so it would only have shown up if the player had landed on Aspiration again.

My fix is to delete the two invisible missions and put the logs into the C05 mission conversation.

One more bug!

Terry's last name was spelled differently in these missions than in the other ones that set her log, leading to two people named Terry with slightly different spellings of their last name but a similar life history.

## Testing Done
Replayed part 6 a few times.

Just before mission C01:
[Friction Diction~3026-05-13 TRY12 B14 before C01.txt](https://github.com/endless-sky/endless-sky/files/10087883/Friction.Diction.3026-05-13.TRY12.B14.before.C01.txt)

Before C02:
[Friction Diction~3026-05-15 TRY12 before C02.txt](https://github.com/endless-sky/endless-sky/files/10087887/Friction.Diction.3026-05-15.TRY12.before.C02.txt)

Before C03:
[Friction Diction~3026-05-22 TRY12 before C03.txt](https://github.com/endless-sky/endless-sky/files/10087889/Friction.Diction.3026-05-22.TRY12.before.C03.txt)

Before C04:
[Friction Diction~3026-05-29 TRY12 before C04.txt](https://github.com/endless-sky/endless-sky/files/10087890/Friction.Diction.3026-05-29.TRY12.before.C04.txt)

Before C05:
[Friction Diction~3026-06-09 TRY12 C05 before end.txt](https://github.com/endless-sky/endless-sky/files/10087891/Friction.Diction.3026-06-09.TRY12.C05.before.end.txt)

## Performance Impact
N/A

### CODE REVIEW

**Comment by UnorderedSigh** [comment]
Hoping for a review from @bene-dictator again. 

There isn't much to review on this one, though.

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:53]:
```suggestion
			`	Apparently, since the Syndicate is technically part of the Republic, they are awarded no special diplomatic status and have been busy with corporate matters elsewhere. Xilin finds you between proceedings and warns you, "Terry seems to have disappeared from Syndicate communications, though I won't hazard a guess as to their internal politics." A warning that carries no surprise at this point.`
```
Maybe this could be made better than my suggestion.

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:65]:
```suggestion
				`	"What's wrong?"`
					goto need2talk
				`	"Hey, don't worry. Whatever it is, I'm sure we've been through worse."`
			`	Sayari rolls her eyes and says, "Maybe so, but there is one more minor disaster to be averted." You follow her to a private meeting room.`
				goto whut
			label need2talk
			`	"We need to talk," says Xilin.`
			`	You feel the deepest of sighs settle into your bones, but at this point you just follow them to a private meeting room. It would have been too easy if it was all over, wouldn't it?`
			label whut
```
This could be rearranged, instead of telling the player what they feel. "We need to talk" could come first, then have the player ask what's wrong.

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:344]:
```suggestion
				`	(Spare her.) "I've sinned more than you. If people like us decided what is right or wrong, the galaxy would be in cinders."`
```

**Comment by bene-dictator** [comment]
I never got the logs for sparing/killing Terry after I finished, could you please check that?

File: data/hai/hai reveal 6 end.txt
Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt:65]: I changed it to "Trouble. We need to talk,"

**Comment by UnorderedSigh** [comment]
> I never got the logs for sparing/killing Terry after I finished, could you please check that?

That is fixed now. See my PR comment for the explanation. Thank you for catching that!

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:66]: I still think it should go like this: 
>As you leave the summit for lunch, Xilin and Sayari intercept you and seem quite worried. 
>"Trouble. We need to talk," says Xilin.

>"What's wrong?" goto whut
>"Hey, don't worry. Whatever it is, I'm sure we've been through worse."

>Sayari rolls her eyes and says, "Maybe so, but there is one more minor disaster to be averted." You follow her to a private meeting room.

>label whut
>Xilin begins. "It seems that somewhere along the way, a shipload of kidnapped Hai didn't make it to a processing area. The Hai were left on Sunracer, and the ship appears to have disappeared." `

Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt:66]: In the "whut" branch, Xilin needs to take you to a private meeting room.

Comment (bene-dictator) [reply data/hai/hai reveal 6 end.txt:66]: Yes, I just realised that.

Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: I rearranged the text and omitted the part about where to talk. Does it work now?

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:55]:
```suggestion
			`	As you leave the summit for lunch, Xilin and Sayari intercept you.`
			`	"Trouble. We need to talk," says Xilin. You follow them to a private meeting room.`
```
Looks better, still could be just a bit more improved. I don't like telling the player what they feel, better for other characters to express that. I'd like to give the player an option to feel frustrated, though, and let Sayari or Xilin talk about "how easy it would be for it to be over".

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:60]:
```suggestion
			`	Sayari rolls her eyes and says, "Maybe so, but there is one more minor disaster to be averted."`
```

File: data/hai/hai reveal 6 end.txt
Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt:55]: > You follow them to a private meeting room.

I'm trying not to lengthen an already quite long scene by adding another paragraph. It doesn't say where you are when the conversation occurs, so it is reasonable to assume you're in a place where you can discuss such matters. Sayari has been a diplomat longer than a human lifespan, so she'd have the experience to know when they can safely talk.

Comment (bene-dictator) [reply data/hai/hai reveal 6 end.txt:55]: I'd imagine you're leaving the Borealis Chamber the moment Sayari and Xilin meet you again, then they would bring you down to the meeting rooms. Otherwise, the player might feel like they're talking out in the open. Only when you get Sayari to roll her eyes, you go to the meeting rooms.

Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: Okay. I put it back in.

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:66]:
```suggestion
				`	"What's wrong?"`
					goto whut
				`	"Really? I thought we had figured everything out."`
				goto need2talk
				`	"Hey, don't worry. Whatever it is, I'm sure we've been through worse."`
			`	Sayari rolls her eyes and says, "Don't relax yet, Captain <last>. There is one more minor disaster to be averted."`
				goto whut
			label need2talk
			`	Sayari says, "It would have been too easy if it was all over. We have one more minor disaster to avert first."`
			label whut
			`	Xilin begins. "It seems that somewhere along the way, a shipload of kidnapped Hai didn't make it to a processing area. The Hai were left on Sunracer, and the ship appears to have disappeared."`
```
While I recommended moving that line to Sayari, it's out of place with what the player says to get to that dialogue, and works better this way.

Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: Your changes were missing a tab. I got them working, though.

**Comment by Saugia** [comment]
Everything good on this, @bene-dictator ?

**Comment by bene-dictator** [comment]
I'll give it one more look.

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:138]: How do the Hai frown in the way they do?

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:359]:
```suggestion
			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself, though, you think: "Maybe the future is worth the sacrifice made in the attempt."`
```
I can't tell what this means. The future is worth the sacrifice?

**Comment by bene-dictator** [comment]
Github doesn't allow me to comment on unedited lines, but I'd change line 155. I didn't know why Korban would send me a message (after all I caused his resignation and I don't know if Korban would take it so... magnanimously, if there's a worse word for it), or even if he was relevant, even if he was the mentor to Terry. This didn't seem to be clear to me, and I think it might need some more explanation, probably better from before this message. I don't know how to do that yet.

File: data/hai/hai reveal 6 end.txt
Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: I just removed the redundant information that the Hai is frowning in the way that Hai frown, since we know Hai always frown in the way that Hai frown.

**Comment by UnorderedSigh** [comment]
> Github doesn't allow me to comment on areas of unedited lines, but I'd change line 155. I didn't know why Korban would send me a message (after all I caused his resignation and I don't know if Korban would take it so forgivingly, or even if he was relevant, even if he was the mentor to Terry. This didn't seem to be clear to me, and I think it might need some more explanation, probably better from before this message. I don't know how or where to do that yet.

I don't think we even know if he's still in power. Now, it comes from "Syndicate Corporate Headquarters" and seems less personal.

**Comment by UnorderedSigh** [comment]
> I'll give it one more look.

To show that everything is good on this, you need to do a review that approves the PR. (You said about a week ago you're new to Github, so I don't know if you are aware of that.)

File: data/hai/hai reveal 6 end.txt
Comment (MasterOfGrey) [inline data/hai/hai reveal 6 end.txt:53]: I'm not sure I like the entirety of this first line swap. It feels like Xilin is acting like he knows things, when if he's warning you it'd at best be a "don't go stick your foot in it over there" type deal.

File: data/hai/hai reveal 6 end.txt
Comment (MasterOfGrey) [inline data/hai/hai reveal 6 end.txt:138]: Remember how earlier we described how Hai naturally are somewhat more alien than you might expect from encountering those familiar with humans?
This was meant to be a subtle nod to the possibility that you've seen enough of them to interpret them sometimes rather than have them adapt to you all the time.

File: data/hai/hai reveal 6 end.txt
Comment (MasterOfGrey) [inline data/hai/hai reveal 6 end.txt:155]: Yep, this is an outdated line, but I think it should actually come from someone you know, either Moray or Remington, leaning towards Moray.

File: data/hai/hai reveal 6 end.txt
Comment (MasterOfGrey) [inline data/hai/hai reveal 6 end.txt:302]: Updating for consistency.
```suggestion
			`	Your arrival is unobstructed as you land on this mild, habitable world that appears to be in an advanced, but retreating, ice age. No one comes to greet you at the spaceport, either. As you wander into the single major town, built on a promontory overlooking a wide river mouth, you notice that the people here seem to be happy and the town appears to be remarkably modern. This place has been looked after by someone who knew how to source the necessary import goods to support it on this sparse world of grasses and something resembling pine trees - someone who was Chief of Operations of the Syndicate, for example...`
```

File: data/hai/hai reveal 6 end.txt
Comment (MasterOfGrey) [inline data/hai/hai reveal 6 end.txt:359]: Can I summon @EjoThims to this specific line? I'm thinking maybe to avoid the clunky punctuation we should go more like:
```suggestion
			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself though, you wonder if perhaps the future is worth the sacrifice made in the attempt.`
```

**Review by MasterOfGrey — CHANGES_REQUESTED** [review]
The mechanical changes look good. Much cleaner now that `action` is a thing.
Otherwise only a handful of thoughts.

File: data/hai/hai reveal 6 end.txt
Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: I used Moray since he shows up less, and thus is less likely to cause complications in future changes to the story.

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [reply data/hai/hai reveal 6 end.txt:138]: Perhaps, but I felt it was not very meaningful unless it was described to the player. I don't know how a Hai frowns, anyway.

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [reply data/hai/hai reveal 6 end.txt:359]: Yup. This was the best I could do with this line, but what does it mean?

File: data/hai/hai reveal 6 end.txt
Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: I rearranged it again. How does it look now?

Comment (MasterOfGrey) [reply data/hai/hai reveal 6 end.txt]: Yes I like that

File: data/hai/hai reveal 6 end.txt
Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt:138]: For it to be meaningful, we would have to describe some aspects of the Hai facial features, but the game is very careful not to describe the various alien species in detail. Rather, we leave it to the player's imagination.

Comment (MasterOfGrey) [reply data/hai/hai reveal 6 end.txt:138]: Which is why we describe such things like familiarity with those foreign details in abstract terms.
I'm not overly hung up on this point though, just felt I should clarify that there was intention to it.
If people don't like it it can stay removed.

Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: I just made a change. How about this? "At this point, he twists his face in a non-human manner you recognize as a Hai frown."

File: data/hai/hai reveal 6 end.txt
Comment (EjoThims) [reply data/hai/hai reveal 6 end.txt]: @UnorderedSigh The most recent suggestion is much better punctuation, but it is still a confusing wording. Especially in reference to which sacrifice and attempt.

I get what it’s referring to, just not what it’s trying to say about what it’s referring to.

Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: The sentence confuses me too, so I've been wary of touching it.

Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: > The most recent suggestion is much better punctuation

I already applied the most recent suggestion.

**Review by EjoThims — CHANGES_REQUESTED** [review]

File: data/hai/hai reveal 6 end.txt
Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt:359]: Yes. The particular morality is already in the `choice`, so it makes sense to remove it here.

File: data/hai/hai reveal 6 end.txt
Comment (MasterOfGrey) [inline data/hai/hai reveal 6 end.txt:342]: Since the whole point is that she's not actually hunted, you're the only person who knows so long as she stays hidden, it's not really appropriate to say she's hunted.
```suggestion
				`	(Spare her.) "There is no other home for you now. Let that be your justice."`
```

File: data/hai/hai reveal 6 end.txt
Comment (MasterOfGrey) [inline data/hai/hai reveal 6 end.txt:340]: I would suggest putting these back in because I think the scene is potentially most powerful with no words said. I think the players should have that option.
```suggestion
				`	"Am I to be judge, jury, and executioner?"`
					goto judge
				`	(Spare her silently.)`
					goto spare
				`	(End her wordlessly.)`
					goto end
					to display
```

File: data/hai/hai reveal 6 end.txt
Comment (MasterOfGrey) [inline data/hai/hai reveal 6 end.txt:344]:
```suggestion
				`	(End her.) "Justice is the price you pay."`
```

File: data/hai/hai reveal 6 end.txt
Comment (MasterOfGrey) [inline data/hai/hai reveal 6 end.txt:348]: I suggest this change because it emphasises that the first two spoken choices are moralising options, whereas the second two are personal options.

I consider this an optional suggestion though.
```suggestion
				`	(End her.) "This is for all the trouble you've caused me, personally."`
```

**Review by MasterOfGrey — CHANGES_REQUESTED** [review]
Ok, very sorry to do this but I spent some time sitting with the new choice options for the end and have some thoughts.

I am definitely good with everything else though.

File: data/hai/hai reveal 6 end.txt
Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt:341]: I restored both of the silent options and updated the `to display` logic, which your suggested change broke. There are *way* too many options, so I changed which ones occur before and after asking if you're the judicial system.

Before asking if you're the judicial system, you get the silent options and the "I don't feel like thinking about it" options:

1. Am I to be judge, jury, and executioner?
2. Silent responses
3. This is for all the trouble you've caused me, personally.
4. Movie review responses

After asking if you're the judicial system, you lose the movie review options and gain the "I thought about it" options:

1. Silent responses
2. There is no other home for you now. Let that be your justice.
3. Justice is the price you pay.
4. I've sinned more than you... wrong... galaxy... cinders.
5. This is for all the trouble you've caused me, personally.

Comment (MasterOfGrey) [reply data/hai/hai reveal 6 end.txt:341]: I like it.

**Review by MasterOfGrey — APPROVED** [review]

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:126]: Purely syntax changes. 
```suggestion
				`	"Well, let's get you home before anything else happens."`
```

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:379]:
```suggestion
			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself, though, you cannot bring yourself to be her executioner.`
```

File: data/hai/hai reveal 6 end.txt
Comment (bene-dictator) [inline data/hai/hai reveal 6 end.txt:398]: Alastair no longer says this.

**Review by bene-dictator — COMMENTED** [review]
Syntax cleanup.

**Comment by Saugia** [comment]
Right when I was about to hit the merge button.,,

**Comment by bene-dictator** [comment]
Sorry!

**Comment by Saugia** [comment]
I've told Grey that from now on, for merges, I need approvals from *both* of you. If you both don't have checkmarks, I won't look at it. This is getting a little silly. :P

**Comment by bene-dictator** [comment]
That's fair, I'm sorry I keep snatching it away from merging! But hopefully this won't take more than 5 minutes to fix up.

File: data/hai/hai reveal 6 end.txt
Comment (UnorderedSigh) [reply data/hai/hai reveal 6 end.txt]: Fixed.

**Comment by bene-dictator** [comment]
Ha ha ha, Grey's only a couple hours from me, whenever Grey's awake I'll be awake.

**Review by bene-dictator — APPROVED** [review]

**Comment by UnorderedSigh** [comment]
@MasterOfGrey Punctuation changes, and a corrected name: Moray is the one who asked Captain Last to warn Terry she would be fired.

**Pull request merged**
